### PR TITLE
Switch to setting up reference tests manually instead of using testset plugin

### PR DESCRIPTION
**Pull request by ajsutton**
Testsets plugin has an incompatibility with IntelliJ and errorprone which prevent compilation in IntelliJ: https://youtrack.jetbrains.com/issue/IDEA-225694

Also remove a few unnecessary dependencies for integration and testSupport where the module has neither.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -316,15 +314,29 @@ subprojects {
     testImplementation 'org.junit.jupiter:junit-jupiter-params'
 
     testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
+
+    integrationTestImplementation 'org.assertj:assertj-core'
+    integrationTestImplementation 'org.apache.tuweni:tuweni-junit'
+    integrationTestImplementation 'org.junit.jupiter:junit-jupiter-api'
+    integrationTestImplementation 'org.junit.jupiter:junit-jupiter-params'
```
Comment (ajsutton) [inline build.gradle:317]: Provide same default class path for integration tests and unit tests.

File: build.gradle [hunk 2]
```diff
@@ -316,15 +314,29 @@ subprojects {
...
   }
 
-  task integrationTest(type: Test, dependsOn:["compileTestJava"]){
+  task integrationTest(type: Test, dependsOn:["compileIntegrationTestJava"]){
     group = "verification"
     description = "Runs the Artemis integration tests"
 
     testClassesDirs = sourceSets.integrationTest.output.classesDirs
     classpath = sourceSets.integrationTest.runtimeClasspath
-    outputs.upToDateWhen { false }
```
Comment (ajsutton) [inline build.gradle:323]: Allow the normal upToDate check to function so we don't re-run tests if nothing has changed (matching behaviour for unit tests).

File: eth-reference-tests/build.gradle [hunk 3]
```diff
@@ -10,6 +10,7 @@ dependencies {
   referenceTestImplementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml'
   referenceTestImplementation 'org.miracl.milagro.amcl:milagro-crypto-java'
   referenceTestImplementation 'org.apache.tuweni:tuweni-bytes'
+  referenceTestImplementation 'org.apache.tuweni:tuweni-junit'
```
Comment (ajsutton) [inline eth-reference-tests/build.gradle:13]: Not sure how we got away without this previously...

**Review by jrhea — APPROVED** [review]
lgtm

**Pull request merged**